src/web_interface/file_tree/file_tree.py

```diff
@@ -86,8 +86,14 @@ def remove_virtual_path_from_root(root: List[dict]) -> List[dict]:
 
 class VirtualPathFileTree:
     '''
-    This class represents a layer or the  file tree  with a ``Firmware`` or ``FileObject`` as root, directories as
-    inner vertices and  ``FileObject``s as outer vertices. It consists of vertices of type ``FileTreeNode``.
+    This class represents a layer of the file tree (a partial tree) for a ``Firmware`` or ``FileObject`` as root and
+    is based on the virtual file paths of its child objets (unpacked files). This tree may have directories as
+    inner vertices (the inner elements of the virtual file path) and ``FileObject``s as outer vertices (the end of the
+    virtual file path). It consists of vertices of type ``FileTreeNode``.
+
```

Comment:
I'd like to add some of your review comment in here as well

Reply:
Please feel free to do so. I feel like I'm too involved to break it down to an understandable level.

Reply:
How 'bout

```python3
    '''
    This class represents a layer of the file tree (a partial tree) for a ``Firmware`` or ``FileObject`` as root and
    is based on the virtual file paths of its child objets (unpacked files). "Layer" means that the file tree is created
    in layers as it is unfolded (one partial tree for each file).

    This partial layer tree has a ``Firmware`` or ``FileObject`` as root, directories as inner nodes (the inner elements
    of the virtual file path) and ``FileObject``s as outer nodes ("leaves" of the tree, the end of the virtual file
    path).

    Both ``Firmware`` and ``FileObject`` vertices are represented by ``FileTreeNode`` objects.

    :param root_uid: The uid of the root node of the file tree.
    :param fo_data: The firmware / file object data from the database that is needed to create the file tree.
    :param whitelist: A whitelist of file names needed to display partial trees in comparisons.
    '''
```

Reply:
Yes, that sounds better 😄